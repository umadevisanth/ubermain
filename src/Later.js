import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

const Later = ({ message, onAddPaymentPress, onDoItLaterPress }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.message}>{message}</Text>
      <TouchableOpacity style={[styles.button, styles.addButton]} onPress={onAddPaymentPress}>
        <Text style={styles.buttonText}>Add Payment Method Now</Text>
      </TouchableOpacity>
      <TouchableOpacity style={[styles.button, styles.doItLaterButton]} onPress={onDoItLaterPress}>
        <Text style={styles.buttonText}>Do It Later</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    padding: 20,
    borderRadius: 10,
  },
  message: {
    marginBottom: 10,
    fontSize: 16,
    textAlign: 'center',
    color:'white'
  },
  button: {
    padding: 10,
    borderRadius: 5,
    marginBottom: 10,
    alignItems: 'center',
  },
  buttonText: {
    color: 'white',
    fontSize: 16,
  },
  addButton: {
    backgroundColor: '#007AFF',
  },
  doItLaterButton: {
    backgroundColor: '#DDDDDD',
  },
});

export default Later;
