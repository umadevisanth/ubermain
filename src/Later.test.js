import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import Later from '../src/Later';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('First component', () => {
  it('renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<Later />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});