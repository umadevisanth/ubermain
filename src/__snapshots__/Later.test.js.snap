// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "black",
      "height": "50%",
      "top": "45%",
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 24,
          "height": 84,
          "left": 44,
          "lineHeight": 28.13,
          "top": 46,
          "width": 341,
        }
      }
    >
      You wonâ€™t be able to request a ride without adding a payment method
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "#EDF6FF",
          "borderWidth": 1,
          "left": 37,
          "opacity": 1,
          "padding": 10,
          "top": 130,
          "width": 300,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Add payment method now
      </Text>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "#EDF6FF",
          "borderWidth": 1,
          "left": 37,
          "opacity": 1,
          "padding": 10,
          "top": 150,
          "width": 300,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Do it later
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
